core/controllers/acl_decorators.py

```diff
@@ -45,6 +47,24 @@
 (suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
 
 
+def handle_redirection(handler, redirection_url, expected_return_type):
```

Comment:
The concept of "permanent" should probably be in both the function name and docstring.